# Modelo de Nível de Encomenda - Sistema $(C_1, C_2)$

## Exercício 3.5

**_O inventário de um artigo, de custo 5 euros por unidade, deve ser controlado utilizando uma política de nível de encomenda._**

**_A procura anual é estimada em 1200 artigos, com uma distribuição da procura durante o prazo de entrega indicada na tabela seguinte._**

| Procura no Prazo de Entrega | 43   | 44   | 45   | 46   | 47   | 48   | 49   | 50   | 51   | 52   | 53   | 54   | 55   | 56   | 57   |
|-----------------------------|------|------|------|------|------|------|------|------|------|------|------|------|------|------|------|
| $p(x)$                      | 0.01 | 0.02 | 0.03 | 0.05 | 0.08 | 0.11 | 0.13 | 0.14 | 0.13 | 0.11 | 0.08 | 0.05 | 0.03 | 0.02 | 0.01 |

**_Se a taxa anual de posse de inventário for $30%$ do preço de compra, e o custo de passagem de encomenda for de $14.10$ euros, determine os parâmetros da referida política nos seguintes casos._**

$b:$ valor unitário do artigo

$r:$ procura anual

$i:$ taxa de juro

$C_1:$ custo de existência

$C_2:$ custo ocurrência de quebra

$C_3:$ custo de passagem de encomenda

In [1]:
b = 5 # euros/unidade
r = 1200 # artigos/ano
i = 0.3 # %/ano

In [2]:
C_1 = i * b; C_1 # euros / artigo / ano

1.5

In [3]:
C_3 = 14.1 # euros / encomenda

Calcular a probabilidade de haver quebra por ciclo de encomenda: P[DDLT > S]

$P[DDLT > S] = P[DDLT > S + 1] + P[DDLT = S + 1]$

Calcular a esperança de quebra: E[DDLT > S]

$E[DDLT > S] = E[DDLT > S + 1] + P[DDLT > S]$

Criar a tabela:

| S | P[DDLT = S] | P[DDLT > S] | E[DDLT > S] |
|-|-|-|-|
| 50 | | | |
| 51 | | | |
| 52 | | | |
| 53 | | | |
| 54 | | | |
| 55 | | | |
| 56 | | | |
| 57 | | | |

1. A gestão especifica que o artigo não deve sofrer quebra de inventário mais do que uma vez por ano.

$QEE = q^{*} = \sqrt{ \frac{2rC_3}{C_1} }$

In [4]:
from math import sqrt, floor
q = sqrt((2 * r * C_3) / C_1);

floor(q) # unidades / encomenda

150

Número médio de situações de quebra por ano: $\frac{r}{q} \cdot P[DDLT > S]$

Número médio de situações de quebra por ano deve ser menor do que 1!

$\frac{r}{q} \cdot P[DDLT > S] \leq 1$

In [5]:
round(1 * (q / r), 4) # P[DDLT > S]

0.1252

Determinal qual o valor de nível de encomenda (S) que probabilidade de quebra é inferior a $0.125$? $S=53$ (consulta da tabela)

**R:** deverão ser encomendadas **150** unidades (q) e o Nível de encomenda deverá ser de **53** (S) para que o artigo **não sofra quebra mais do que 1x no ano**.

2. Estudos indicam que cada vez que há quebra, se verifica um custo fixo de 4 euros.

In [6]:
C_2 = 4

3. Estudos indicam que se verifica um custo de 1 euro por cada artigo em atraso.